tasks/dist.js

```diff
@@ -0,0 +1,106 @@
+module.exports = function(grunt) {
+
+    // location of our temporary working copy we use to build tags
+    var tempCheckout = 'temp/checkout';
+
+    // register a few promise related convenience functions
+    var exec = require('./utils')(grunt).exec;
```

Comment:
You could at least also remove the `rm -rf` and replace one `cp`. I would avoid shelling out as much as possible (the only cases where you have to is when you call grunt again or want to use the globbing).